### PR TITLE
network: add support for MACVLAN source mode

### PR DESCRIPTION
**Pull request by sipraga**
Add support for creating a MACVLAN interface in "source" mode by
specifying Mode=source in the [MACVLAN] section of a .netdev file.

A list of allowed MAC addresses for the corresponding MACVLAN can also
be specified with the SourceMACAddress= option of the [MACVLAN] section.

An example .netdev file:

    [NetDev]
    Name=macvlan0
    Kind=macvlan
    MACAddress=02:DE:AD:BE:EF:00

    [MACVLAN]
    Mode=source
    RemoteMACAddress=02:AB:AB:AB:AB:01 02:CD:CD:CD:CD:01
    RemoteMACAddress=02:EF:EF:EF:EF:01

The same keys can also be specified in [MACVTAP] for MACVTAP kinds of
interfaces, with the same semantics.

### CODE REVIEW

**Comment by sipraga** [comment]
Thanks for the comments. I have made the requested changes and force pushed. Please take a look.

**Comment by yuwata** [comment]
LGTM.

File: man/systemd.netdev.xml [hunk 1]
```diff
@@ -499,11 +499,22 @@
...
+          <literal>passthru</literal>, and
+          <literal>source</literal>.
           </para>
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><varname>SourceMACAddress=</varname></term>
+        <listitem>
+          <para>A whitespace-separated list of remote hardware addresses allowed on the MACVLAN. This
+          option only has an affect in source mode. Use full colon-, hyphen- or dot-delimited
```
Comment (keszybz) [inline man/systemd.netdev.xml:512]: affect → effect

**Comment by keszybz** [comment]
bionic-i386 timed out.

**Pull request merged**